src/content_sao.cpp

```diff
@@ -1102,13 +1102,12 @@ int PlayerSAO::punch(v3f dir,
 
 	if (!damage_handled) {
 		setHP(getHP() - hitparams.hp);
-	} else { // override client prediction
-		if (puncher->getType() == ACTIVEOBJECT_TYPE_PLAYER) {
-			std::string str = gob_cmd_punched(0, getHP());
-			// create message and add to list
-			ActiveObjectMessage aom(getId(), true, str);
-			m_messages_out.push(aom);
-		}
+	} else if (puncher && puncher->getType() == ACTIVEOBJECT_TYPE_PLAYER) {
```

Comment:
If so, the other NULL checks should be removed, but I assumed it can be NULL since it's checked for NULL in a few places in this function.